### ISSUE TITLE
obj and mtl importing

### ISSUE THREAD

**Issue by jyy1082** [WhitestormJS/whitestorm.js#363]
Does whs.js support to import obj and mtl files with image resource? If yes, is there anyone can guide me how to use WHS.Importer to add those into WHS.App? 

Thank you!

###### Version:
 - [x] v2.1.9

###### Issue type:
 - [ ] Bug
 - [ ] Proposal/Enhancement
 - [X ] Question
 - [ ] Discussion

------

###### Desktop
   - [X ] Chrome

**Comment by sasha240100** [comment]
Hey, @jyy1082 , sorry for delayed answer.
**TL;DR:** It's possible, and here's how:

### Importing obj
```js
const object = new WHS.Importer({
  loader: new THREE.OBJLoader(), // you need to provide obj loader there
  // ... 
});
```

### Importing mtl
You may write your own module, which can be reusable then, but the simplest way is to use [three-mtl-loader](https://www.npmjs.com/package/three-mtl-loader) and then:
```js
mtlLoader.load('model.mtl', function(matl) {
  const object = new WHS.Importer({
    material: matl
    // ... 
  });

  // ...
});
```

And the final stage, add this to app:

```js
object.addTo(app);
```

**Comment by jyy1082** [comment]
Hi, 
Thanks for your replay!
My obj and mtl files are for one model. Can I import them together as a object using one WHS.Importer?

**Comment by sasha240100** [comment]
multiple obj files, right?

**Comment by jyy1082** [comment]
I mean one obj and one mtl files together for one model.

https://www.learnthreejs.com/load-3d-model-using-three-js-obj-loader/

This is the way to use THREEJS to implement. How should I use WHS to do that? Thanks.

**Comment by sasha240100** [comment]
@jyy1082 See the code above, it's pretty similar with what you need

**Comment by jyy1082** [comment]
I will try it. Thanks again!

**Comment by hirako2000** [comment]
does it work @jyy1082 ?

**Comment by antoniocapelo** [comment]
I'm also struggling a bit with this. 
This snippet works when using with vanilla ThreeJS loader:

```
        var loader = newTHREE.OBJLoader();
        loader.load( 'o.obj', ( object ) => {
            this.scene.add( object );
        });

```

However, when doing the 'same' thing with WHS Importer: 

```
        new WHS.Importer({
            url: 'o.obj',
            loader: new OBJLoader(),

            parser(geometry, material) { // material comes as undefined since i'm not loading any .mtl file
                return new THREE.Mesh(geometry, new THREE.MeshNormalMaterial({
                    color: 0xff0000,
                }));
            },

            position: [0, 100, 0]
        }).addTo(this.app);

```

I got an error in a three.module.js codebase `TypeError: Cannot read property 'center' of undefined
    at Sphere.copy (three.module.js?8bc5:8345)`

![image](https://user-images.githubusercontent.com/3236388/35439182-868f15d4-0291-11e8-91ef-72e5599f4768.png)


Currently I'm using the vanilla approach but would like how to integrate better with WhiteStorm.
Cheers

**Comment by sasha240100** [comment]
@antoniocapelo, since you’ve moved to OBJLoader, your parser() accept other arguments.

In three.js you have: `loader.load(‘o.js’, parser);` and in your code instead of `parser` you have an arrow function with object as input data. Same in whs. 

parser arguments:
JSONLoader: (geometry, material)
OBJLoader: (object)

Should look like this:

```js
parser(object) {
  return parser;
}
```

**Comment by sasha240100** [comment]
```js
parser(object) {
  return object;
}
```

Correct one. (made a typo in my prev msg)

**Comment by antoniocapelo** [comment]
Sorry for the delay @sasha240100 ! I tried what you described and it does work (not when loading the `.mtl` file at the same time though — But the 'vanilla' way I described is looking good until now, so that's covered).

Thanks!

**Comment by noahcoetsee** [comment]
@sasha240100 I am once again volunteering myself to create a more permanent module for loading these models in (unless someone's already done it?).

**Comment by sasha240100** [comment]
I could share MTLModule source with you and maybe we could easily combine
it with obj

**Comment by noahcoetsee** [comment]
Yeah, that sounds great! Where's the source?

**Comment by sasha240100** [comment]
@TheCodeCrafter Sorry for late response. I wasn't able to find the source code (probably lost it because of not using git for that project). I could help you in development, just point me

**Comment by noahcoetsee** [comment]
Ok, I will try to remember to do this in the near future. My programming world is a hellish nightmare of projects (abandoned and current) everywhere right now. Will return when ready, sir! ;)

**Comment by zardilior** [comment]
This issue should be closed as it looks solved

**Comment by hirako2000** [comment]
Agreed @zardilior but it would be nice to add an example to illustrate how to do it, anyone volunteers to add a simple example?

**Comment by noahcoetsee** [comment]
I'm unable to at the moment, as I'm super busy with school. I'll get back to you during late December!
